platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/Configsets.java

```diff
@@ -80,11 +84,61 @@ private Path createDefaultConfigset(Path configsets) {
 
   public Path get(String collection) {
     Path collectionPath = Paths.get(configsetsPath, collection, "conf");
-    if (collectionPath.toFile().exists()) {
-      return collectionPath;
-    } else {
+    if (!collectionPath.toFile().exists()) {
       LOGGER.debug("No configset for collection [{}]. Using default configset instead", collection);
       return defaultPath;
     }
+    return getRectifiedConfig(collection, collectionPath);
+  }
+
+  private Path getRectifiedConfig(String collection, Path collectionPath) {
+    // if any defaults are missing-- create a temp  and copy collectionPath + defaults
+    Set<Path> allFiles =
+        SOLR_CONFIG_FILES.stream()
+            .map(configName -> Paths.get(collectionPath.toString(), configName))
+            .collect(Collectors.toSet());
+    Set<Path> missingFiles =
+        allFiles.stream().filter(config -> !config.toFile().exists()).collect(Collectors.toSet());
+    Set<Path> existingFiles =
+        allFiles.stream().filter(missingFiles::contains).collect(Collectors.toSet());
+
+    if (missingFiles.isEmpty()) {
+      return collectionPath;
+    }
+
+    Path tempConfigDir;
+    try {
+      tempConfigDir = Files.createTempDirectory("solr-configsets");
+      tempConfigDir.toFile().deleteOnExit();
+    } catch (IOException e) {
```

Comment:
❔ If something else goes wrong in there do you need to make sure that tempConfigDir is closed?

Reply:
The nice thing about using a temp directory is though we might make a couple of them (if multiple attempts are made or if something goes wrong), since its a temp directory the OS will clean it out (at a minimum) when restarted. The overhead isn't very high, so it should never accumulate into a problem.  The delete on exit is just an optimistic way to clean it out possibly sooner.